scripts/update-chart.sh

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env bash
+
+# update-chart.sh handles creating a new version of the parent chart, updating
+# its dependencies and pushing the chart into the repository.
+
+# In order to compare chart dependencies, we transform the information into a
+# JSON map of service to version. Then we can use jq, which has more powerful
+# comparison operators than bash to determine the difference.
+
+set -Eeo pipefail
+
+USAGE="
+Usage:
+  $(basename "$0")
+
+Environment variables:
+  COMMIT_TARGET_BRANCH: Name of the GitHub branch where the committed changes
+                        will be pushed. Defaults to 'main'
+  GITHUB_ORG:           Name of the GitHub organisation where committed changes
+                        to the chart will be pushed.
+                        Defaults to 'aws-controllers-k8s'
+  GITHUB_REPO:          Name of the GitHub repository where committed changes to
+                        the chart will be pushed.
+                        Defaults to 'ack-parent-chart'
+  GITHUB_TOKEN:        Personal Access Token for '$GITHUB_ACTOR'
+"
+
+SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT_DIR="$SCRIPTS_DIR/.."
+WORKSPACE_DIR="$ROOT_DIR/.."
+
+TEST_INFRA_LIB_DIR="$WORKSPACE_DIR/test-infra/scripts/lib"
```

Comment:
If we are making an assumption that `test-infra` is installed when someone runs this script, that assumption needs to be called out in the README

However, do we actually need it?  Can you snip out the necessary parts of `common.sh` and import them into this repo?

Reply:
These scripts are only meant to be run by the CD system, which will be configured in Prow the same way that our other scripts are - that is, they have copies of all the other common ACK packages in their correct respective paths